services/impl/src/main/java/org/exoplatform/automatic/translation/rest/AutomaticTranslationRestService.java

```diff
@@ -53,57 +54,31 @@ public AutomaticTranslationRestService(AutomaticTranslationService automaticTran
   @Path("/configuration")
   @RolesAllowed("administrators")
   @Produces(MediaType.APPLICATION_JSON)
-  @ApiOperation(
-      value = "Gets Automatic Translation configuration",
-      httpMethod = "GET",
-      response = Response.class,
-      produces = MediaType.APPLICATION_JSON,
-      notes = "This returns the actual configuration for automatic translation"
-  )
-  @ApiResponses(
-      value = {
-          @ApiResponse(code = 200, message = "Request fulfilled")
-      }
-  )
+  @ApiOperation(value = "Gets Automatic Translation configuration", httpMethod = "GET", response = Response.class, produces = MediaType.APPLICATION_JSON, notes = "This returns the actual configuration for automatic translation")
+  @ApiResponses(value = { @ApiResponse(code = 200, message = "Request fulfilled") })
   public Response configuration() {
-    JSONObject resultJSON = new JSONObject();
-
     String activeConnector = automaticTranslationService.getActiveConnector();
-    JSONArray connectors = new JSONArray();
+    AutomaticTranslationConfiguration configuration = new AutomaticTranslationConfiguration(new ArrayList<>(), activeConnector);
```

Comment:
For better performances and resources consuming to cache the configuration object and compute it in Service layer, so that we don't have to instantiate multiple objects each time we call this REST endpoint. i.e : add a new method in Service layer to return the configuration object that is shared and managed in-memory.

Reply:
done